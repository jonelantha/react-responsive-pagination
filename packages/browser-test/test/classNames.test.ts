import { setupThrowOnError, stringifyWithUndefined } from './helper';

beforeAll(async () => {
  setupThrowOnError(page);

  await page.goto(`${harnessUrl}bootstrap4`);

  await page.setViewportSize({ width: 700, height: 700 });
});

beforeEach(async () => {
  await page.fill('#classNameAsJson', 'undefined');
  await page.fill('#extraClassNameAsJson', 'undefined');
  await page.fill('#pageItemClassNameAsJson', 'undefined');
  await page.fill('#pageLinkClassNameAsJson', 'undefined');
  await page.fill('#activeItemClassNameAsJson', 'undefined');
  await page.fill('#disabledItemClassNameAsJson', 'undefined');
  await page.fill('#srOnlyClassNameAsJson', 'undefined');
});

describe('Setting extraClassName', () => {
  test.each([
    [undefined, 'pagination justify-content-center'],
    ['', 'pagination'],
    ['justify-content-start', 'pagination justify-content-start'],
  ])('will have expected class', async (extraClassProp, expectedFullClass) => {
    await page.fill('#extraClassNameAsJson', stringifyWithUndefined(extraClassProp));

    await page.evaluate(() => new Promise(requestAnimationFrame));

    const pagination = await page.waitForSelector('ul.pagination');

    const fullClass = await pagination.getAttribute('class');

    expect(fullClass).toBe(expectedFullClass);
  });
});

describe('classNames', () => {
  test.each([
    ['className', 'alt-pagination'],
    ['pageItemClassName', 'alt-page-item'],
    ['pageLinkClassName', 'alt-page-link'],
    ['activeItemClassName', 'alt-active'],
    ['disabledItemClassName', 'alt-disabled'],
    ['srOnlyClassName', 'alt-sr-only'],
  ])('setting %p to %p renders correctly', async (field, value) => {
    await page.fill(`#${field}AsJson`, JSON.stringify(value));

    await page.evaluate(() => new Promise(requestAnimationFrame));

    const paginationHtml = await page.$eval('ul', ul => ul.outerHTML);
    expect(paginationHtml).toMatchSnapshot();
  });

  test('setting "srOnlyClassName" to "" removes screen reader spans', async () => {
    await page.fill(`#srOnlyClassNameAsJson`, JSON.stringify(''));

    await page.evaluate(() => new Promise(requestAnimationFrame));

    const paginationHtml = await page.$eval('ul', ul => ul.outerHTML);
    expect(paginationHtml).toMatchSnapshot();
  });
});
