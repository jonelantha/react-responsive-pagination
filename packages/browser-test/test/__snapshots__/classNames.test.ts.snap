// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classNames setting "activeItemClassName" to "alt-active" renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item alt-active"><a class="page-link" href="#">1<span class="sr-only"> (current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`classNames setting "className" to "alt-pagination" renders correctly 1`] = `"<ul class="alt-pagination"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#">1<span class="sr-only"> (current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`classNames setting "disabledItemClassName" to "alt-disabled" renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item alt-disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#">1<span class="sr-only"> (current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item alt-disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`classNames setting "pageItemClassName" to "alt-page-item" renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="alt-page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="alt-page-item active"><a class="page-link" href="#">1<span class="sr-only"> (current)</span></a></li><li class="alt-page-item"><a class="page-link" href="#">2</a></li><li class="alt-page-item"><a class="page-link" href="#">3</a></li><li class="alt-page-item"><a class="page-link" href="#">4</a></li><li class="alt-page-item"><a class="page-link" href="#">5</a></li><li class="alt-page-item"><a class="page-link" href="#">6</a></li><li class="alt-page-item"><a class="page-link" href="#">7</a></li><li class="alt-page-item"><a class="page-link" href="#">8</a></li><li class="alt-page-item"><a class="page-link" href="#">9</a></li><li class="alt-page-item"><a class="page-link" href="#">10</a></li><li class="alt-page-item"><a class="page-link" href="#">11</a></li><li class="alt-page-item"><a class="page-link" href="#">12</a></li><li class="alt-page-item"><a class="page-link" href="#">13</a></li><li class="alt-page-item"><a class="page-link" href="#">14</a></li><li class="alt-page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="alt-page-item"><a class="page-link" href="#">100</a></li><li class="alt-page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`classNames setting "pageLinkClassName" to "alt-page-link" renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="alt-page-link" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="alt-page-link" href="#">1<span class="sr-only"> (current)</span></a></li><li class="page-item"><a class="alt-page-link" href="#">2</a></li><li class="page-item"><a class="alt-page-link" href="#">3</a></li><li class="page-item"><a class="alt-page-link" href="#">4</a></li><li class="page-item"><a class="alt-page-link" href="#">5</a></li><li class="page-item"><a class="alt-page-link" href="#">6</a></li><li class="page-item"><a class="alt-page-link" href="#">7</a></li><li class="page-item"><a class="alt-page-link" href="#">8</a></li><li class="page-item"><a class="alt-page-link" href="#">9</a></li><li class="page-item"><a class="alt-page-link" href="#">10</a></li><li class="page-item"><a class="alt-page-link" href="#">11</a></li><li class="page-item"><a class="alt-page-link" href="#">12</a></li><li class="page-item"><a class="alt-page-link" href="#">13</a></li><li class="page-item"><a class="alt-page-link" href="#">14</a></li><li class="page-item"><a class="alt-page-link" href="#">15</a></li><li class="page-item"><a class="alt-page-link" href="#">16</a></li><li class="page-item"><a class="alt-page-link" href="#">17</a></li><li class="page-item disabled" aria-hidden="true"><span class="alt-page-link">…</span></li><li class="page-item"><a class="alt-page-link" href="#">100</a></li><li class="page-item"><a class="alt-page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`classNames setting "srOnlyClassName" to "" removes screen reader spans 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`classNames setting "srOnlyClassName" to "alt-sr-only" renders correctly 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span><span class="alt-sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#">1<span class="alt-sr-only"> (current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="alt-sr-only">Next</span></a></li></ul>"`;
